spec/duration.html

```diff
@@ -1167,81 +1331,292 @@ <h1>
+        1. If _one_.[[TotalNanoseconds]] &lt; _two_.[[TotalNanoseconds]], return -1.
+        1. Return 0.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-dividenormalizedtimeduration" type="abstract operation">
+      <h1>
+        DivideNormalizedTimeDuration (
+          _d_: a Normalized Time Duration Record,
+          _divisor_: an integer,
+        ): a mathematical value
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It divides the total number of nanoseconds in the normalized time duration _d_ by _divisor_.</dd>
+      </dl>
+      <emu-alg>
+        1. Assert: _divisor_ &ne; 0.
+        1. Return _d_.[[TotalNanoseconds]] / _divisor_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-normalizedtimedurationfromepochnanosecondsdifference" type="abstract operation">
+      <h1>
+        NormalizedTimeDurationFromEpochNanosecondsDifference (
+          _one_: a BigInt,
+          _two_: a BigInt,
+        ): a Normalized Time Duration Record
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It creates a Normalized Time Duration Record with the difference between two exact times in nanoseconds since the epoch, which must not be greater than the maximum normalized time duration.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _result_ be ℝ(_one_) - ℝ(_two_).
+        1. Assert: abs(_result_) &le; maxTimeDuration.
+        1. Return the Record { [[TotalNanoseconds]]: _result_ }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-normalizedtimedurationiszero" type="abstract operation">
+      <h1>
+        NormalizedTimeDurationIsZero (
+          _d_: a Normalized Time Duration Record,
+        ): a Boolean
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns *true* if _d_ is a zero duration, *false* otherwise.</dd>
+      </dl>
+      <emu-alg>
+        1. If _d_.[[TotalNanoseconds]] = 0, return *true*.
+        1. Return *false*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-roundnormalizedtimedurationtoincrement" type="abstract operation">
+      <h1>
+        RoundNormalizedTimeDurationToIncrement (
+          _d_: a Normalized Time Duration Record,
+          _increment_: an integer,
+          _roundingMode_: *"ceil"*, *"floor"*, *"expand"*, *"trunc"*, *"halfCeil"*, *"halfFloor"*, *"halfExpand"*, *"halfTrunc"*, or *"halfEven"*,
+        ): a Normalized Time Duration Record
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It rounds the total number of nanoseconds in the normalized time duration _d_ to the nearest multiple of _increment_, up or down according to _roundingMode_.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _rounded_ be RoundNumberToIncrement(_d_.[[TotalNanoseconds]], _increment_, _roundingMode_).
+        1. Return the Record { [[TotalNanoseconds]]: _rounded_ }.
```

Comment:
The returned value isn't guaranteed to be a valid normalised time duration. 

```js
let d = Temporal.Duration.from({
  seconds: 2**53 - 1,
});
let r = d.round({
  smallestUnit: "seconds",
  roundingIncrement: 30,
  roundingMode: "ceil",
});
```

Reply:
You're right, I think the solution is this operation just needs to be able to throw.

Reply:
I don't _think_ that would affect Duration.p.total because RoundDuration is called from there with roundingIncrement 1 and roundingMode trunc.